### DISCUSSION TITLE
why am i getting same response from my GET function

### DISCUSSION THREAD

**Discussion by rtolzo** [vercel/next.js#59222] [Help] [answered]
### Summary

my code returns time at the requested moment

```js
import { NextResponse } from "next/server";

export async function GET() {
  return NextResponse.json({ time: new Date().toLocaleString() });
}

```

and every time I request, I get the same result
Does anybody know why??

**Comment by wslp12** [comment]
Your app is on the route cache and needs to be hard refreshed to operate properly.
There are several ways to solve this, generally, run router.refresh() at the client level. Of course, your fetch also needs to be revalidated.

**Comment by icyJoseph** [comment, marked as answer]
Hi,

In App Router, Router handlers using the GET method, under certain conditions, get automatically statically optimized.

You can opt-out of this in many ways.

For example:

```ts
import { NextResponse } from "next/server";

export const dynamic = 'force-dynamic';

export async function GET() {
  return NextResponse.json({ time: new Date().toLocaleString() });
}
```

Other ways include using the incoming request object. Read more here: https://nextjs.org/docs/app/building-your-application/routing/route-handlers#opting-out-of-caching

**Reply by rtolzo** [reply]
this solution works! thanks a lot:)
